pkg/ddl-checker/executable_checker.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl_checker
+
+import (
+	"context"
+	"fmt"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/pingcap/tidb/util/logutil"
+)
+
+type ExecutableChecker struct {
+	session  session.Session
+	context  context.Context
```

Comment:
try to avoid putting the context in the structure,  the best way to use context is to initialize one that accompanies the process at the beginning of the process